src/main/java/de/tum/cit/aet/artemis/core/service/telemetry/EurekaClientService.java

```diff
@@ -0,0 +1,92 @@
+package de.tum.cit.aet.artemis.core.service.telemetry;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Collections;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@Service
+public class EurekaClientService {
```

Comment:
Could you add the `@Profile` annotation here?

---

```
Rule 'classes that are annotated with @Service or are annotated with @Component or are annotated with @Configuration and do not belong to any of [de.tum.cit.aet.artemis.core.config.ApplicationConfiguration, de.tum.cit.aet.artemis.core.config.ConditionalMetricsExclusionConfiguration] should be annotated with @Profile, because we want to be able to exclude these classes from application startup by specifying profiles' was violated
```